docs/components/reference.mdx

```diff
@@ -0,0 +1,1180 @@
+
+| Type | Name                                                                | Description                                                | Style       | Usage          |
+|------|---------------------------------------------------------------------|------------------------------------------------------------|-------------|----------------|
+| 1    | [Action Row](/docs/components/reference#action-row)                 | Container to display a row of interactive components       | Layout      | Message, Modal |
+| 2    | [Button](/docs/components/reference#button)                         | Button object                                              | Interactive | Message        |
+| 3    | [String Select](/docs/components/reference#string-select)           | Select menu for picking from defined text options          | Interactive | Message        |
+| 4    | [Text Input](/docs/components/reference#text-input)                 | Text input object                                          | Interactive | Modal          |
+| 5    | [User Select](/docs/components/reference#user-select)               | Select menu for users                                      | Interactive | Message        |
+| 6    | [Role Select](/docs/components/reference#role-select)               | Select menu for roles                                      | Interactive | Message        |
+| 7    | [Mentionable Select](/docs/components/reference#mentionable-select) | Select menu for mentionables (users *and* roles)           | Interactive | Message        |
+| 8    | [Channel Select](/docs/components/reference#channel-select)         | Select menu for channels                                   | Interactive | Message        |
+| 9    | [Section](/docs/components/reference#section)                       | Container to display text alongside an accessory component | Layout      | Message        |
+| 10   | [Text Display](/docs/components/reference#text-display)             | Markdown text                                              | Content     | Message        |
+| 11   | [Thumbnail](/docs/components/reference#thumbnail)                   | Small image that can be used as an accessory               | Content     | Message        |
+| 12   | [Media Gallery](/docs/components/reference#media-gallery)           | Display images and other media                             | Content     | Message        |
+| 13   | [File](/docs/components/reference#file)                             | Displays an attached file                                  | Content     | Message        |
+| 14   | [Separator](/docs/components/reference#separator)                   | Component to add vertical padding between other components | Layout      | Message        |
+| 17   | [Container](/docs/components/reference#container)                   | Container that visually groups a set of components         | Layout      | Message        |
```

Comment:
I assume we decided against including the content inventory type as reference since it can be received?